code/datums/quirks/good.dm

```diff
@@ -36,19 +35,20 @@
 	gain_text = "<span class='notice'>You feel like a drink would do you good.</span>"
 	lose_text = "<span class='danger'>You no longer feel like drinking would ease your pain.</span>"
 	medical_record_text = "Patient has unusually efficient liver metabolism and can slowly regenerate wounds by drinking alcoholic beverages."
+	processing_quirk = TRUE
 
-/datum/quirk/drunkhealing/on_process(delta_time)
-	var/mob/living/carbon/C = quirk_holder
-	switch(C.drunkenness)
+/datum/quirk/drunkhealing/process(delta_time)
```

Comment:
Adding onto above about quirks controlling their own processing, you could improve the performance of this by only processing when the quirk holder does get drunk and then stop processing when they're no longer drunk.

Reply:
I don't really fancy micro-optimising quirk code that hard.

Something still needs to process in order to decide when the carbon starts being drunk and that means either sending the signal every life tick the carbon is drunk (this would be the solution in the current code), or adding new state tracking to every carbon to send a signal when their drunkenness state changes.

You'd probably find your micro-optimisation gains rapidly lost in that sea for a simple quirk. If a maintainer disagrees I'd be happy to implement it in another PR.

Reply:
It's up to you in the end. Though you could easily send a signal whenever drunkenness changes and then make a setter proc for it.

Reply:
That is more likely a refactor for another PR.

Reply:
The suggestion is good, and would be welcome on a different PR.